docs/language/informal/covariant-overrides.md

```diff
@@ -0,0 +1,742 @@
+code:
+
+```dart
+var group = new RadioGroup();
+group.addChild(new Widget());
+```
+
+There is no static error even though it's obviously statically incorrect.
+
+Anyone who has designed a widget hierarchy has likely run into this problem a
+couple of times. In particular, this showed up in a number of places in Flutter.
+
+In some cases, you can solve this using generics:
+
+```dart
+class Widget<T extends Widget> {
+  void addChild(T widget) {...}
+}
+
+class RadioButton extends Widget<Null> {
+  void select() {...}
+}
+
+class RadioGroup extends Widget<RadioButton> {
+  void addChild(RadioButton button) {
+    button.select();
+    super.addChild(button);
+  }
+}
+```
+
+In practice, this often doesn't work out. Often you have a family of related
+types, and making one generic means they all have to be, each with type
+parameters referring to the other. It often requires API consumers to make their
+own code generic simply to pass these objects around.
+
+## The covariant override feature
+
+Earlier, we showed the pattern users manually apply when they want to tighten a
+parameter type in a method override:
+
+```dart
+void addChild(Widget widget) {
+  var button = widget as RadioButton;
+  ...
+}
+```
+
+This proposal is roughly akin to syntactic sugar for that pattern. In this
+method, `widget` effectively has *two* types:
+
+*   The **original type** is the type that is declared on the method parameter.
+    Here, it's `Widget`. This type must follow the type system's rules for a
+    valid override in order to preserve the soundness guarantees of the
+    language.
+
+    With method parameters, that means the type must be equivalent to or a
+    supertype of the parameter type in the superclass and all of its
+    superinterfaces. This is the usual
+    [sound rule for function subtyping][contra].
+
+[contra]: https://en.wikipedia.org/wiki/Covariance_and_contravariance_(computer_science)#Function_types
+
+*   The **desired type** is the type that the user wants to use inside the body
+    of the method. This is the real type that the overridden method requires in
+    order to function. Here, it's `RadioButton`.
+
+    The desired type is a subtype of the original type. Going from the original
+    type to the desired type is a downcast, which means the cast needs to be
+    checked and may fail at runtime.
+
+    Even though the desired type is ensconced in the body of the method in the
+    manual pattern, it really is part of the method's signature. If you are
+    calling `addChild()` *on an object that you statically know is a
+    RadioGroup*, you want the errors as if its declared type was the tighter
+    desired type, not the original one. This is something the manual pattern
+    can't express.
+
+So we need to understand the original type, the desired type, and when
+this feature comes into play. To enable it on a method parameter, you
+mark it with the contextual keyword `covariant`:
```

Comment:
If possible, yes. That's the least breaking way to introduce it.

The alternative is to make it a built-in identifier like `get` and `set`.